src/Components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -466,20 +479,21 @@ const ResourcePicker: React.FC<IResourcePickerProps> = ({
                         setOptions(newOptions);
                         setSelectedOption(newOption);
                         setSelectedKey(newParsedOptionValue);
+
+                        if (onChange)
+                            onChange(
+                                newParsedOptionValue,
+                                getResourcesFromOptions(newOptions) || []
+                            );
+                    } else {
+                        setSelectedOption(existingOption);
+                        setSelectedKey(existingOption.key);
```

Comment:
why do we need to store both the option and the key? That seems like a bad smell.

Reply:
because it is a controlled combobox where you can add additional values with freeform or with changes in the prop for selected option I need to be able to set and control the selected key: https://developer.microsoft.com/en-us/fluentui#/controls/web/combobox (Controlled ComboBox)